lib/matplotlib/rcsetup.py

```diff
@@ -310,6 +318,25 @@ def validate_colorlist(s):
         msg = "'s' must be of type [ string | list | tuple ]"
         raise ValueError(msg)
 
+def validate_cycler(s):
+    'return a cycler object from a string repr or the object itself'
+    if isinstance(s, six.string_types):
+        try:
+            # TODO: We might want to rethink this...
+            cycler_inst = eval(s, {'cycler': cycler, 'Cycler': Cycler})
```

Comment:
It already does contain code, `val = ['a', 'b', 'c']` is code.  Also see
lisp rabbit hole ;)

On Sat, Aug 15, 2015 at 2:07 AM OceanWolf <notifications@github.com> wrote:

> In lib/matplotlib/rcsetup.py
> <https://github.com/matplotlib/matplotlib/pull/4686#discussion_r37133851>:
>
> > @@ -310,6 +318,25 @@ def validate_colorlist(s):
> >          msg = "'s' must be of type [ string | list | tuple ]"
> >          raise ValueError(msg)
> >
> > +def validate_cycler(s):
> > +    'return a cycler object from a string repr or the object itself'
> > +    if isinstance(s, six.string_types):
> > +        try:
> > +            # TODO: We might want to rethink this...
> > +            cycler_inst = eval(s, {'cycler': cycler, 'Cycler': Cycler})
>
> Sure, I see that, it just feels messy, a config file shouldn't contain
> code...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4686/files#r37133851>.
>